src/trs/rewrite.rs

```diff
@@ -117,10 +128,82 @@ impl TRS {
             .sum()
     }
 
+    /// test the likelihood
+    pub fn test_single_likelihood(&self, datum: &Rule, params: ModelParams) {
```

Comment:
The only reason this function is here is because `TRS.utrs` is `pub(crate)` rather than pub. I'll remove it from the PR.

I suggested implementing `From<TRS> for UTRS` as a way around the `pub(crate)` restriction in #21.

Another option is to merge UTRS and TRS. The main difference is in their grammaticality checkers (i.e. `Signature` and `Lexicon`). `Signature` looks at arity, while `Lexicon` looks at types. Developing a common interface would allow for a single `TRS` type in the `term_rewriting` crate and dramatically simplify what's needed in `programinduction`.

Reply:
This was resolved in 02ca6071021100d64c4a8a023b94223ae321535e.